### PR TITLE
Backend dev search history

### PR DESCRIPTION
**Pull request by HUmutOzdemir**

### CODE REVIEW

File: Platon/backend/app/search_engine/views.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        form = SearchHistoryForm(request.args)
+        if form.validate():
+            if form.search_type.data > 2 or form.search_type.data < 0:
+                return make_response(jsonify({"error": "Search Type must be 0, 1 or 2"}), 400)
+            try:
+                search_history =  db.engine.execute("SELECT `query` FROM `search_history_item` WHERE `user_id`={} AND `type`={}".format(user_id,form.search_type.data))           
+            except:
+                return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+            search_hist_dict = {}
+            for history_item in search_history:
```
Comment (cakmadam98) [inline Platon/backend/app/search_engine/views.py:178]: Is there any specific advantages of sorting search history with respect to number of times the query performed?

**Review by cakmadam98 — APPROVED** [review]
Great 💯

File: Platon/backend/app/search_engine/views.py (hunk 1 above)
Comment (HUmutOzdemir) [reply Platon/backend/app/search_engine/views.py:178]: Think about a case a user searches for same query for so many times. For these cases I have to remove duplicates. So , while removing duplicates, I also sorting them according the their usaga rate. There is not any specification about them in requirements. So I made the decission.

**Review by alperendivriklioglu — APPROVED** [review]
I'm a bit unfamiliar with the mechanics of semantic search, so, I could not get into each detail; but I could not notice any major problem either. So, I think we can merge this code.

**Pull request merged**